pkg/chain/ethereum/beacon.go

```diff
@@ -0,0 +1,30 @@
+package ethereum
+
+import (
+	"context"
+	"fmt"
+	"github.com/keep-network/keep-common/pkg/chain/ethereum"
+	"github.com/keep-network/keep-common/pkg/chain/ethereum/ethutil"
+)
+
+// BeaconChain represents a beacon-specific chain handle.
+type BeaconChain struct {
+	*Chain
+}
+
+// NewBeaconChain construct a new instance of the beacon-specific Ethereum
+// chain handle.
+func NewBeaconChain(
+	ctx context.Context,
+	config *ethereum.Config,
+	client ethutil.EthereumClient,
+) (*BeaconChain, error) {
+	chain, err := NewChain(ctx, config, client)
```

Comment:
The problem with creating a new instance of the base chain in `New*` function of the application chain is that we will end up with multiple instances of the base chain: one for Random Beacon and one for TBTC. If the base chain holds the transaction mutex - and this feels like the right place - it means we have two mutexes.

To avoid it, I would consider keeping the base chain as a singleton and creating application chains using public functions of the base chain. 

I did this change in https://github.com/keep-network/keep-core/pull/3067/commits/a3ba22abc7ce4b0c75413b261a1e11d12df8361c in the followup PR of #3067.

Reply:
Please also see https://github.com/keep-network/keep-core/pull/3067/commits/fa755027dde397dda86b1dfd428c4fa271cbf88a that deals with `Connect` function.